src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -42,15 +42,18 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             var cancellationToken = context.CancellationToken;
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-            var identifier = await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start,
-                token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
+            var identifier = context.Span.IsEmpty
+                ? await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start, token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false)
+                : await root.SyntaxTree.GetLeftmostTokenInSpanAsync(context.Span, token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
+
+
             if (identifier == default)
             {
                 return;
             }
 
             if (context.Span.Length > 0 &&
-                context.Span != identifier.Span)
+                (context.Span != identifier.Span && context.Span != identifier.Parent.Span))    // either local function's identifier Token or the whole function's span
```

Comment:
everything is && so, () probably not needed?

Reply:
so, if caret is on right token, or that token is selected or whole node is selected, we offer refactoring. seems reasonable. 

caret on the same line as the identifier offering refactoring feels too noisy? like what fix does?

Reply:
1) Yep, it's not needed. Just a readability preference that shows logical grouping. Can remove. 
2) If carret is only on the same line and not directly next to identifier then we don't offer refactoring. The logic should be as follows: 
   -  No selection: offer refactorings only when carrot directly next to identifier
   -  Selection: Offer only when either the whole method or the identifier is selected.